src/Datadog.Trace/Agent/Api.cs

```diff
@@ -20,14 +19,20 @@ internal class Api : IApi
 
         private readonly IApiRequestFactory _apiRequestFactory;
         private readonly IDogStatsd _statsd;
-        private readonly FormatterResolverWrapper _formatterResolver = new FormatterResolverWrapper(SpanFormatterResolver.Instance);
         private readonly string _containerId;
         private readonly Uri _tracesEndpoint;
+        private readonly IDatadogTracer _tracer;
         private string _cachedResponse;
 
         public Api(Uri baseEndpoint, IApiRequestFactory apiRequestFactory, IDogStatsd statsd)
+            : this(baseEndpoint, apiRequestFactory, statsd, tracer: null)
+        {
+        }
+
+        internal Api(Uri baseEndpoint, IApiRequestFactory apiRequestFactory, IDogStatsd statsd, IDatadogTracer tracer)
```

Comment:
This is already an `internal` class. We usually make members `public` in cases like this.
```suggestion
        public Api(Uri baseEndpoint, IApiRequestFactory apiRequestFactory, IDogStatsd statsd, IDatadogTracer tracer)
```

Reply:
I looked into this some more and I think this overload is only used for testing. Can you add a comment to that effect? Thanks.

Reply:
I find the circular dependency weird (`Tracer` -> `IAgentWriter` -> `IApi` and now `Api` -> `IDatadogTracer`), which made think some more.

I'm wondering if `ReportAgentVersion()` should be on `Api` instead of `Tracer`, since communicating to the Agent falls within `Api`'s responsibilities and not `Tracer`'s. In theory, `Tracer` shouldn't know how or where traces are going.

Reply:
> I looked into this some more and I think this overload is only used for testing. Can you add a comment to that effect? Thanks.

Sure

> I find the circular dependency weird (Tracer -> IAgentWriter -> IApi and now Api -> IDatadogTracer), which made think some more.

The dependency is not new, we're already calling `Tracer.Instance.Sampler.SetDefaultSampleRates(apiResponse?.RateByService);` in Api.

Reply:
Oh, well... this is fine for now, then. Maybe someday we can come back and clean this all up.